WordPress/Classes/Services/RawBlockEditorSettingsService.swift

```diff
@@ -24,48 +25,50 @@ class RawBlockEditorSettingsService {
             guard let dictionary = response as? [String: Any] else {
                 throw NSError(domain: "RawBlockEditorSettingsService", code: 1, userInfo: [NSLocalizedDescriptionKey: "Invalid response format"])
             }
-            blog.rawBlockEditorSettings = dictionary
+            let objectID = TaggedManagedObjectID(blog)
+            try? await ContextManager.shared.performAndSave { context in
+                let blog = try context.existingObject(with: objectID)
+                blog.setBlockEditorSettings(dictionary)
+            }
             return dictionary
         case .failure(let error):
             throw error
         }
     }
 
+    /// Refreshes the editor settings in the background.
+    func refreshSettings() {
+        Task { @MainActor in
+            try? await fetchSettings()
```

Comment:
The new `refreshSettings` method does not decode the editor settings. It used to happen every time you open the "My Site" screen and it was relatively expensive, and was happening on the main thread. We don't need to read the current settings to send a request to refresh them.

Reply:
Avoiding unnecessary decoding is smart.